app/concepts/todo/create.rb

```diff
@@ -1,50 +1,22 @@
 class Todo::Create < Trailblazer::Operation
+  include Model
+  model Todo
+  
+  # validate if the title is blank
+  contract do
+    property :title, validates: {presence: true}
+  end
 
-  # Logic for creating a TODO
-  #     if `todo_list` is specified
-  #         create the `todo`
-  #         associated with `todo_list`
-  #     else
-  #         create the `todo`
-  #         create a `user`
-  #         create a default `todo_list`
-  #         associate `todo_list` with `user`
-  #         associate `todo` with `todo_list`
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
-
-    # build the todo
+    
     todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
-
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
-
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
-        else
-          params[:current_user]
-        end
-
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
+    
+    validate(params[:todo], @model) do |f|
+      todo_list = params[:todo_list_id].blank? ? TodoList::Create.(params).model : TodoList.find(params[:todo_list_id])
+      todo.title = f.title
+      todo.list = todo_list
+      todo.save
     end
-
-    # associate the todo with the todo_list
-    todo.list = todo_list
-    todo.save!
   end
 
 end
```

Comment:
The code is definitely simpler. It's wonderful to look at. Almost perfect.

Can you reflect on this code and 

```
class Todo::Create < Trailblazer::Operation
    include Model
    model Todo, :create

    contract do
        property :name
        property :todo_list

        validates :name, presence: true
    end

    def process(params)
        validate(params[:todo]) do |f|
            f.todo_list = find_todo_list(params)
            f.save
        end
    end

    private

    def find_todo_list(params)
        TodoList::Find.(params).model
    end
end
```

As you can see, `Todo::Create` does not know what `TodoList::Find` does in the background. It only trusts that calling `.model` after the operation is run returns the correct `todo_list`. It's the responsibility of `TodoList::Find` to find/create the proper `todo_list` given the passed `params`.

We can say that this code adheres to the single responsibility principle. This class has one and only one purpose - to create a `todo` that belongs to a `todo_list`. Nothing more, nothing less.

The code I just showed you is not yet perfect because the class depends on the existence of a class named `TodoList::Find`. What if we delete that class? Then this class needs to be modified for the code to work. However for the purposes of this coding challenge, it's good enough.

Can you modify your other operations to look like this? Our senior dev isn't a fan of comments so if you can move all description comments to an RSpec spec instead, that would be great.